### PR TITLE
Take into account learning rate of composites and recurrents

### PR DESCRIPTION
**Pull request by FedyuninV**

### CODE REVIEW

File: NeoML/test/src/DnnSolverTest.cpp [hunk 1]
```diff
@@ -813,3 +829,64 @@ TEST( CDnnLambGradientSolverTest, Serialization4 )
 	lamb->SetWeightDecayClip( 2.5f );
 	solverSerializationTestImpl( lamb.Ptr(), true );
 }
+
+// ====================================================================================================================
+
+static void checkBlobInequality( CDnnBlob& firstBlob, CDnnBlob& secondBlob )
```
Comment (lordnn) [inline NeoML/test/src/DnnSolverTest.cpp:835]: Why don't reuse `checkBlobEquality()`?

File: NeoML/test/src/DnnSolverTest.cpp [hunk 2]
```diff
@@ -813,3 +829,64 @@ TEST( CDnnLambGradientSolverTest, Serialization4 )
...
+{
+	ASSERT_TRUE( firstBlob.HasEqualDimensions( &secondBlob ) );
+	const int dataSize = firstBlob.GetDataSize();
+	float* first = firstBlob.GetBuffer<float>( 0, dataSize, true );
+	float* second = secondBlob.GetBuffer<float>( 0, dataSize, true );
+	bool success = false;
+	for( int i = 0; i < dataSize; ++i ) {
+		success |= !FloatEqImpl( first[i], second[i], 1e-4f );
```
Comment (lordnn) [inline NeoML/test/src/DnnSolverTest.cpp:843]: After first `true` you may skip rest checks.

Comment (FedyuninV) [reply NeoML/test/src/DnnSolverTest.cpp:843]: It will be removed #869

File: NeoML/test/src/DnnSolverTest.cpp (hunk 1 above)
Comment (FedyuninV) [reply NeoML/test/src/DnnSolverTest.cpp:835]: #869

**Pull request merged**